pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -100,6 +152,11 @@ func (v *storagePolicyAPI) createStoragePolicy(ctx context.Context) (string, err
 		return v.policyName, nil
 	}
 
+	vSphereInfraConfig := v.infra.Spec.PlatformSpec.VSphere
+	if vSphereInfraConfig != nil && len(vSphereInfraConfig.FailureDomains) > 1 {
+		return v.createZonalStoragePolicy(ctx)
+	}
+
```

Comment:
`createZonalStoragePolicy` and rest of `createStoragePolicy` below are almost the same, `createZonalStoragePolicy` only loops / tags over more datastores. Would it be possible to join them together somehow? Like `getZonalDatastores()` + `getDefaultDatastore()` and then a common tagging loop + `createStorageProfile` over the result.

Reply:
okay I have unified that code. But I have pushed my new changes to https://github.com/openshift/vmware-vsphere-csi-driver-operator/pull/125/files#diff-cd720ec01eefb8566ee378bc53aef398da5550c725aeacd65aba7efb2b39e311R146 

That branch already includes commit from this branch and has more tests and ensures tags are recreated if deleted etc.